### PR TITLE
Use the correct type for values in PropertiesChanged 

### LINKED ISSUE #115: Wrong type of structs on PropertiesChanged
In D-Installer we have an object with the following properties:

```
‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=191  ReplyCookie=207  Timestamp="Tue 2022-06-14 12:14:39.530914 UTC"
  Sender=:1.1175  Destination=:1.1171
  UniqueName=:1.1175
  MESSAGE "a{sv}" {
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "RootPasswordSet";
                          VARIANT "b" {
                                  BOOLEAN false;
                          };
                  };
                  DICT_ENTRY "sv" {
                          STRING "RootSSHKey";
                          VARIANT "s" {
                                  STRING "";
                          };
                  };
                  DICT_ENTRY "sv" {
                          STRING "FirstUser";
                          VARIANT "(ssba{sv})" {
                                  STRUCT "ssba{sv}" {
                                          STRING "";
                                          STRING "user1";
                                          BOOLEAN false;
                                          ARRAY "{sv}" {
                                          };
                                  };
                          };
                  };
          };
  };
```

However, when we launch a `PropertiesChanged` signal to change the `FirstUser` property, the list of changes looks like this:

```
‣ Type=signal  Endian=l  Flags=0  Version=1 Cookie=215  Timestamp="Tue 2022-06-14 12:14:44.260642 UTC"
  Sender=:1.1175  Path=/org/opensuse/DInstaller/Users1  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.1175
  MESSAGE "sa{sv}as" {
          STRING "org.opensuse.DInstaller.Users1";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "FirstUser";
                          VARIANT "av" {
                                  ARRAY "v" {
                                          VARIANT "s" {
                                                  STRING "";
                                          };
                                          VARIANT "s" {
                                                  STRING "user1";
                                          };
                                          VARIANT "b" {
                                                  BOOLEAN false;
                                          };
                                          VARIANT "a{sv}" {
                                                  ARRAY "{sv}" {
                                                  };
                                          };
                                  };
                          };
                  };
          };
          ARRAY "s" {
          };
  };
```

We expected a struct, but that's not the case. The problem can be reproduced by using the D-Installer CLI:

```
cd service
sudo bundle exec bin/dinstaller &
cd ../cli
bundle exec bin/dinstallerctl user set user1
```

Could you have a look, please? Thanks!

### PR DESCRIPTION
**Pull request by mvidner**
To fix #115

- [x] NEWS.md, and mention how dbus_watcher changes
- [x] test, probably similarly to property_spec in #113

### CODE REVIEW

File: spec/object_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+describe DBus::Object do
+  describe ".dbus_attr_writer" do
+    describe "the declared assignment method" do
+      # Slightly advanced RSpec:
+      # https://rspec.info/documentation/3.9/rspec-expectations/RSpec/Matchers.html#satisfy-instance_method
+      let(:a_struct_in_a_variant) do
+        satisfying { |x| x.is_a?(DBus::Data::Variant) && x.member_type.to_s == "(ss)" }
+        # ^ This formatting keeps the matcher on a single line
+        # which enables RSpect to cite it if it fails, instead of saying "block".
```
Comment (imobachgs) [inline spec/object_spec.rb:23]: np: RSpec

File: spec/property_spec.rb [hunk 2]
```diff
@@ -110,6 +110,7 @@
     end
 
     @iface["ReadOrWriteMe"] = "VALUE"
+    @iface.SetTwoProperties("REAMDE", 255)
```
Comment (imobachgs) [inline spec/property_spec.rb:113]: np: REAMDE or README?

**Review by imobachgs — APPROVED** [review]
I am not an expert, but the change makes sense. And it is also clear what we need to change in D-Installer to make it work. Thanks!

File: spec/property_spec.rb (hunk 2 above)
Comment (mvidner) [reply spec/property_spec.rb:113]: It's a cultural joke :) https://en.wikipedia.org/wiki/Reamde

**Pull request merged**